### PR TITLE
Write project content attributes to projects directly

### PR DESCRIPTION
**Pull request by marten**
Describe your change here.

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -62,6 +62,12 @@ def update
       # transaction.
       content_attributes = primary_content_attributes(update_params)
       unless content_attributes.blank?
+        resource.display_name = content_attributes[:title] if content_attributes.key?(:title)
```
Comment (marten) [inline app/controllers/api/v1/projects_controller.rb]: I'm not super happy with how this is set up. But couldn't come up with something nicer that worked. Let me hear suggestions if you have m

File: app/controllers/api/v1/projects_controller.rb [hunk 2]
```diff
@@ -62,6 +62,12 @@ def update
...
+        resource.description = content_attributes[:description] if content_attributes.key?(:description)
```
[the remark this answers is not in the document]
Comment (marten) [reply app/controllers/api/v1/projects_controller.rb]: Yeah the display_name vs title was the issue causing one to be weird, and I figured a bunch of lines would be nicer than a loop with one exception. But you're right, that'll get set anyway, which makes a loop easy to do here.

**Comment by marten** [comment]
@camallen is it easier to follow now?

File: app/controllers/api/v1/projects_controller.rb [hunk 3]
```diff
@@ -128,14 +134,15 @@ def build_resource_for_create(create_params)
 
     content_attributes = primary_content_attributes(create_params)
     create_params[:project_contents] = [ ProjectContent.new(content_attributes) ]
+    create_params[:url_labels] = content_attributes[:url_labels]
```
Comment (camallen) [inline app/controllers/api/v1/projects_controller.rb]: It may not do but does this need the associated `urls` attribute as well?

Comment (marten) [reply app/controllers/api/v1/projects_controller.rb]: No, that's updated in place in the original params hash (which is horrible but out of scope unless you want me to clean it up)

**Pull request merged**